dimension/client.go

```diff
@@ -100,37 +141,76 @@ func (c *Client) GetAreaTypes(ctx context.Context, userAuthToken, serviceAuthTok
 		}
 	}()
 
-	if resp.StatusCode == http.StatusNotFound {
-		var errorResp ErrorResp
-		if err := json.NewDecoder(resp.Body).Decode(&errorResp); err == nil {
-			return GetAreaTypesResponse{}, dperrors.New(
-				fmt.Errorf("error response from Dimensions API (%d): %w", resp.StatusCode, errorResp),
-				http.StatusInternalServerError,
-				logData,
-			)
-		}
+	if err := checkDimensionGetResponse(logData, *resp); err != nil {
+		return GetAreaTypesResponse{}, err
 	}
 
-	if resp.StatusCode != http.StatusOK {
-		// Best effort — an empty body is fine for the error message
-		body, _ := io.ReadAll(resp.Body)
+	var areaTypes GetAreaTypesResponse
+	if err := json.NewDecoder(resp.Body).Decode(&areaTypes); err != nil {
 		return GetAreaTypesResponse{}, dperrors.New(
-			fmt.Errorf("error response from Dimensions API (%d): %s", resp.StatusCode, body),
+			fmt.Errorf("unable to deserialize area types response: %w", err),
 			http.StatusInternalServerError,
 			logData,
 		)
 	}
 
-	var areaTypes GetAreaTypesResponse
-	if err := json.NewDecoder(resp.Body).Decode(&areaTypes); err != nil {
-		return GetAreaTypesResponse{}, dperrors.New(
-			fmt.Errorf("unable to deserialize area types response: %w", err),
+	return areaTypes, nil
+}
+
+func (c *Client) GetAreas(ctx context.Context, userAuthToken, serviceAuthToken, datasetID, areaTypeID, text string) (GetAreasResponse, error) {
+	logData := log.Data{
+		"method":       http.MethodGet,
+		"dataset_id":   datasetID,
+		"area_type_id": areaTypeID,
+		"text":         text,
+	}
+
+	urlPath := "areas"
+	urlValues := url.Values{"dataset": []string{datasetID}, "area-type": []string{areaTypeID}}
+	if text != "" {
+		urlValues.Add("text", text)
+	}
```

Comment:
Although an empty string should still be a no-op, it's an optional field so might be worth having the same treatment as `text`.